File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.cc
#include <bits/stdc++.h>
using namespace std;

int main(void){
  ios::sync_with_stdio(0);
  cin.tie(0);

  int n, m;
  cin >> n >> m;

  unordered_map<string, string> hash;
  while(n--){
    string link, password;
    cin >> link >> password;

    hash[link] = password;
  }

  while(m--){
    string link;
    cin >> link;

    cout << hash[link] << '\n';
  }
}

File: 백준/Silver/17219. 비밀번호 찾기/비밀번호 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.StringTokenizer;

public class Main {

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static HashMap<String, String> hash = new HashMap<>();

    public static void main(String[] args) throws IOException {
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        while(n-- > 0){
            st = new StringTokenizer(br.readLine());
            String link = st.nextToken();
            String password = st.nextToken();

            hash.put(link, password);
        }

        while(m-- > 0){
            String password = br.readLine();
            System.out.println(hash.get(password));
        }
    }

}
